src/TestFramework/TestLogger.php

```diff
@@ -0,0 +1,190 @@
+<?php
+
+namespace PaymentGatewayLogger\TestFramework;
+
+use Psr\Log\AbstractLogger;
+
+/**
+ * Pulled from https://github.com/php-fig/log/ modified for php >= 5.3
+ *
+ * Used for testing purposes.
+ *
+ * It records all records and gives you access to them for verification.
+ *
+ * @method bool hasEmergency($record)
+ * @method bool hasAlert($record)
+ * @method bool hasCritical($record)
+ * @method bool hasError($record)
+ * @method bool hasWarning($record)
+ * @method bool hasNotice($record)
+ * @method bool hasInfo($record)
+ * @method bool hasDebug($record)
+ *
+ * @method bool hasEmergencyRecords()
+ * @method bool hasAlertRecords()
+ * @method bool hasCriticalRecords()
+ * @method bool hasErrorRecords()
+ * @method bool hasWarningRecords()
+ * @method bool hasNoticeRecords()
+ * @method bool hasInfoRecords()
+ * @method bool hasDebugRecords()
+ *
+ * @method bool hasEmergencyThatContains($message)
+ * @method bool hasAlertThatContains($message)
+ * @method bool hasCriticalThatContains($message)
+ * @method bool hasErrorThatContains($message)
+ * @method bool hasWarningThatContains($message)
+ * @method bool hasNoticeThatContains($message)
+ * @method bool hasInfoThatContains($message)
+ * @method bool hasDebugThatContains($message)
+ *
+ * @method bool hasEmergencyThatMatches($message)
+ * @method bool hasAlertThatMatches($message)
+ * @method bool hasCriticalThatMatches($message)
+ * @method bool hasErrorThatMatches($message)
+ * @method bool hasWarningThatMatches($message)
+ * @method bool hasNoticeThatMatches($message)
+ * @method bool hasInfoThatMatches($message)
+ * @method bool hasDebugThatMatches($message)
+ *
+ * @method bool hasEmergencyThatPasses($message)
+ * @method bool hasAlertThatPasses($message)
+ * @method bool hasCriticalThatPasses($message)
+ * @method bool hasErrorThatPasses($message)
+ * @method bool hasWarningThatPasses($message)
+ * @method bool hasNoticeThatPasses($message)
+ * @method bool hasInfoThatPasses($message)
+ * @method bool hasDebugThatPasses($message)
```

Comment:
what is all this @method stuff for?

Reply:
i'm a bit confused about how this file works...any clarity would be helpful

Reply:
These `@methods` are due to the `__call` magic method which allows you to call all of the `has*` methods with a the debug level i.e. notice, warn, info and debug, etc. Essentially it's just a way to document `__call`. For example `hasRecords()` can be called in several ways with just the one generic method. Invoking:

```PHP
hasEmergencyRecords()
hasAlertRecords()
hasCriticalRecords(
hasErrorRecords()
hasWarningRecords()
hasNoticeRecords()
hasInfoRecords()
hasDebugRecords()
```

will get intercepted by the `__call` magic method which just parses out the logging type and calls the generic `hasRecords()` function.